server/controllers/users.js

```diff
@@ -52,30 +52,31 @@ export const exists = async (req, res) => {
 /**
  * Login or create a new user
  */
-export const signin = (req, res, next) => {
-  const { user, redirect, websiteUrl } = req.body;
-  let loginLink;
-  let clientIP;
-  return models.User.findOne({ where: { email: user.email.toLowerCase() } })
-    .then(u => u || models.User.createUserWithCollective(user))
-    .then(u => {
-      loginLink = u.generateLoginLink(redirect || '/', websiteUrl);
-      clientIP = req.ip;
-      if (config.env === 'development') {
-        logger.info(`Login Link: ${loginLink}`);
-      }
-      return emailLib.send('user.new.token', u.email, { loginLink, clientIP }, { sendEvenIfNotProduction: true });
-    })
-    .then(() => {
-      const response = { success: true };
-      // For e2e testing, we enable testuser+(admin|member)@opencollective.com to automatically receive the login link
-      if (config.env !== 'production' && user.email.match(/.*test.*@opencollective.com$/)) {
-        response.redirect = loginLink;
-      }
-      return response;
-    })
-    .then(response => res.send(response))
-    .catch(next);
+export const signin = async (req, res) => {
+  const { user, redirect, websiteUrl, createProfile = true } = req.body;
+  let profile = await models.User.findOne({ where: { email: user.email.toLowerCase() } });
+  if (!profile && !createProfile) {
+    return res.send({
+      error: {
+        errorCode: 'EMAIL_DOES_NOT_EXIST',
+        message: 'Email does not exist',
+      },
+    });
+  } else if (!profile && createProfile) {
+    profile = models.User.createUserWithCollective(user);
```

Comment:
Was that tested ?

Reply:
This `else if` never triggered currently from the front-end. The reason is whenever we use `signin` function, we first call `exists` to see the email exists. Only if it exists we call the `signin` function. However I checked if this works correctly with a mock (calling the `signin` function regardless of whether email exists or not). We could leave this for now and once the old signin flow is removed from front-end could refactor this function to only sign-in and remove this collective creation logic. 🤔

Reply:
Yes you are correct, I initially thought the accounts were created using this in the sign-in flow. But then realized that we call a different mutation to create the profiles and this branch wasn't hit at all which is why I initially missed testing this. Sorry about that. 😢